Add VMware documentation (#1)
zmrow/bottlerocket: README.md

Bottlerocket is a free and open-source Linux-based operating system meant for hosting containers.

If you’re ready to jump right in, read our [QUICKSTART for EKS](QUICKSTART-EKS.md) to try Bottlerocket in an Amazon EKS cluster or our [QUICKSTART for Amazon ECS](QUICKSTART-ECS.md) to try Bottlerocket in an Amazon ECS cluster.

> If you'd like to try Bottlerocket as a Kubernetes worker node in VMware, read the [QUICKSTART for VMware](QUICKSTART-VMWARE.md).

Reviewer: ```suggestion
If you’re ready to jump right in, read one of our setup guides for running Bottlerocket in [Amazon EKS](QUICKSTART-EKS.md), [Amazon ECS](QUICKSTART-ECS.md), or [VMware](QUICKSTART-VMWARE.md).
```

Bottlerocket is best used with a container orchestrator.

> To get started with Kubernetes, please see [QUICKSTART-EKS](QUICKSTART-EKS.md).

Reviewer: ```suggestion
To get started with Kubernetes in Amazon EKS, please see [QUICKSTART-EKS](QUICKSTART-EKS.md).
```

To get started with Kubernetes in VMware, please see [QUICKSTART-VMWARE](QUICKSTART-VMWARE.md).
To get started with Amazon ECS, please see [QUICKSTART-ECS](QUICKSTART-ECS.md).
These guides describe:

> * how to set up a cluster with the orchestrator, so your Bottlerocket instance can run containers (for EKS and ECS)

Reviewer: ```suggestion
* how to set up a cluster with the orchestrator, so your Bottlerocket instance can run containers
```

This container runs the [AWS SSM agent](https://github.com/aws/amazon-ssm-agent) that lets you run commands, or start shell sessions, on Bottlerocket instances in EC2.
(You can easily replace this control container with your own just by changing the URI; see [Settings](#settings).)

In AWS, you need to give your instance the SSM role for this to work; see the [setup guide](QUICKSTART-EKS.md#enabling-ssm).

> Outside of AWS, you can use [AWS Systems Manager for hybrid environments](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-managedinstances.html).

Reviewer: ```suggestion
Outside of AWS, you can use [AWS Systems Manager for hybrid environments](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-managedinstances.html).
There's more detail about hybrid environments in the [control container documentation](https://github.com/bottlerocket-os/bottlerocket-control-container/#connecting-to-aws-systems-manager-ssm).
```

### Admin container

Bottlerocket has an [administrative container](https://github.com/bottlerocket-os/bottlerocket-admin-container), disabled by default, that runs outside of the orchestrator in a separate instance of containerd.

> This container has an SSH server that lets you log in as `ec2-user` using your EC2-registered SSH key.  Outside of AWS, you can pass in your [own SSH keys](https://github.com/bottlerocket-os/bottlerocket-admin-container#authenticating-with-the-admin-container).

Reviewer: ```suggestion
This container has an SSH server that lets you log in as `ec2-user` using your EC2-registered SSH key.
Outside of AWS, you can [pass in your own SSH keys](https://github.com/bottlerocket-os/bottlerocket-admin-container#authenticating-with-the-admin-container).
```

#### Kubernetes settings

See the [EKS setup guide](QUICKSTART-EKS.md) for much more detail on setting up Bottlerocket and Kubernetes in AWS EKS.
For more details about running Bottlerocket as a Kubernetes worker node in VMware, see the [VMware setup guide](QUICKSTART-VMWARE.md).

The following settings must be specified in order to join a Kubernetes cluster.
You should [specify them in user data](#using-user-data).
* `settings.kubernetes.cluster-certificate`: This is the base64-encoded certificate authority of the cluster.
* `settings.kubernetes.api-server`: This is the cluster's Kubernetes API endpoint.

> For AWS EKS variants, you must also specify:

Reviewer: ```suggestion
For AWS Kubernetes variants, you must also specify:
```

Reviewer 2: We should avoid the phrase "AWS Kubernetes".

Reviewer: Good point.  `For Kubernetes variants in AWS,`

* `settings.kubernetes.cluster-name`: The cluster name you chose during setup; the [setup guide](QUICKSTART-EKS.md) uses "bottlerocket".

> For VMware variants, you must specify:

Reviewer: ```suggestion
For VMware Kubernetes variants, you must also specify:
```

* `settings.kubernetes.cluster-dns-ip`: The IP of the DNS service running in the cluster.

> * `settings.kubernetes.bootstrap-token`: The token used for [TLS bootstrapping](https://kubernetes.io/docs/reference/command-line-tools-refe    rence/kubelet-tls-bootstrapping/).

Reviewer 2: weird spaces here

* `settings.kubernetes.static-pods.<custom identifier>.manifest`: A base64-encoded pod manifest.
* `settings.kubernetes.static-pods.<custom identifier>.enabled`: Whether the static pod is enabled.

> The following settings are set for you automatically by [pluto](sources/api/) for AWS EKS variants based on runtime instance information, but you can override them if you know what you're doing!

Reviewer: ```suggestion
The following settings are set for you automatically by [pluto](sources/api/) (for AWS Kubernetes variants) based on runtime instance information, but you can override them if you know what you're doing!
```

Reviewer: Seems like we'd want some kind of note about how these work for VMware?
